### PR TITLE
asciidoc apidoc generation

### PR DESCRIPTION
**Pull request by cbosdo**
## What does this PR change?

Add asciidoc API doc generation.

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed: This is an improvement for documentation :)

- [X] **DONE**

## Test coverage
- No tests: tooling

- [X] **DONE**

## Links

Fixes #
Tracks # **add downstream PR, if any**

- [X] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [X] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"		 
- [ ] Re-run test "java_pgsql_tests"		 
- [ ] Re-run test "ruby_rubocop"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"		 
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"		 
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: java/buildconf/ivy/ivy-suse.xml [hunk 1]
```diff
@@ -92,6 +92,7 @@
         <dependency org="suse" name="xerces-j2" rev="2.11.0" />
         <dependency org="suse" name="xmlsec" rev="2.0.7" />
         <dependency org="suse" name="velocity" rev="1.5" />
+        <dependency org="suse" name="commons-lang" rev="2.6" />
```
Comment (mcalmer) [inline java/buildconf/ivy/ivy-suse.xml]: uhhh, we tried so hard to get rid of commons-lang 2 . Is there no posibility to use v3?

File: java/code/internal/src/com/redhat/rhn/internal/doclet/AsciidocDoclet.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) 2019 SUSE LLC
```
Comment (mcalmer) [inline java/code/internal/src/com/redhat/rhn/internal/doclet/AsciidocDoclet.java]: 2020

File: java/code/internal/src/com/redhat/rhn/internal/doclet/AsciidocWriter.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) 2019 SUSE LLC
```
Comment (mcalmer) [inline java/code/internal/src/com/redhat/rhn/internal/doclet/AsciidocWriter.java]: 2020 now.

File: java/buildconf/ivy/ivy-suse.xml (hunk 1 above)
Comment (cbosdo) [reply java/buildconf/ivy/ivy-suse.xml]: we'll need to investigate upgrading velocity then.

File: java/code/internal/src/com/redhat/rhn/internal/doclet/AsciidocDoclet.java (hunk 2 above)
Comment (cbosdo) [reply java/code/internal/src/com/redhat/rhn/internal/doclet/AsciidocDoclet.java]: arf indeed

File: java/buildconf/ivy/ivy-suse.xml (hunk 1 above)
Comment (cbosdo) [reply java/buildconf/ivy/ivy-suse.xml]: Commented out that line

**Pull request merged**